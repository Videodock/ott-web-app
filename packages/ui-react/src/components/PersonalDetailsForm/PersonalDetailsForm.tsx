import React from 'react';
import { useTranslation } from 'react-i18next';
import type { FormErrors } from '@jwp/ott-common/types/form';
import type { CleengCaptureField, CleengCaptureQuestionField, PersonalDetailsFormData } from '@jwp/ott-common/types/account';
import { testId } from '@jwp/ott-common/src/utils/common';

import TextField from '../TextField/TextField';
import Button from '../Button/Button';
import Dropdown from '../Dropdown/Dropdown';
import Checkbox from '../Checkbox/Checkbox';
import Radio from '../Radio/Radio';
import DateField from '../DateField/DateField';
import LoadingOverlay from '../LoadingOverlay/LoadingOverlay';
import FormFeedback from '../FormFeedback/FormFeedback';

import styles from './PersonalDetailsForm.module.scss';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  setValue: (key: keyof PersonalDetailsFormData, value: string) => void;
  error?: string;
  errors: FormErrors<PersonalDetailsFormData>;
  validationError?: boolean;
  values: PersonalDetailsFormData;
  submitting: boolean;
  fields: Record<string, CleengCaptureField>;
  questions: CleengCaptureQuestionField[];
  onQuestionChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  questionValues: Record<string, string>;
  questionErrors: Record<string, string>;
};

const PersonalDetailsForm: React.FC<Props> = ({
  onSubmit,
  onChange,
  setValue,
  validationError,
  values,
  errors,
  submitting,
  fields,
  questions,
  onQuestionChange,
  questionValues,
  questionErrors,
}: Props) => {
  const { t } = useTranslation('account');
  const renderQuestion = ({ value, key, question, required, enabled }: CleengCaptureQuestionField) => {
    if (!enabled) {
      return null;
    }

    const values = value?.split(';').map((question) => {
      const [value, label = value] = question.split(':');

      return { value, label };
    });

    const props = {
      name: key,
      onChange: onQuestionChange,
      error: !!questionErrors[key],
      helperText: questionErrors[key],
      required,
      key,
    };

    // The rendered field is determined by the given available options for each question:
    // TextField <- when there is exactly 1 option and the value is empty (e.g. "")
    // Checkbox  <- when there is exactly 1 option and the value is not empty (e.g. "accepted")
    // Radio     <- when there are exactly 2 options
    // Dropdown  <- when there are more than 2 options

    if (values.length === 1 && values[0].value === '') {
      return <TextField value={questionValues[key]} label={question} {...props} />;
    } else if (values.length === 1) {
      return <Checkbox checked={!!questionValues[key]} value={values[0].value} header={question} label={values[0].label} {...props} />;
    } else if (values.length === 2) {
      return <Radio values={values} value={questionValues[key]} header={question} {...props} />;
    } else if (values.length > 2) {
      return <Dropdown options={values} value={questionValues[key]} label={question} defaultLabel={t('personal_details.select_answer')} {...props} fullWidth />;
    }

    return null;
  };

  return (
    <form onSubmit={onSubmit} data-testid={testId('personal_details-form')} noValidate>
      <h2 className={styles.title}>{t('personal_details.title')}</h2>
      {errors.form ? (
        <FormFeedback variant="error" visible={!validationError}>
          {errors.form}
        </FormFeedback>
      ) : null}
      {fields.firstNameLastName?.enabled ? (
        <React.Fragment>
          <TextField
            value={values.firstName}
            onChange={onChange}
            label={t('personal_details.fist_name')}
            placeholder={t('personal_details.fist_name')}
            error={!!errors.firstName}
            helperText={errors.firstName}
            required={fields.firstNameLastName.required}
            name="firstName"
            autoComplete="given-name"
          />
          <TextField
            value={values.lastName}
            onChange={onChange}
            label={t('personal_details.last_name')}
            placeholder={t('personal_details.last_name')}
            error={!!errors.lastName}
            helperText={errors.lastName}
            required={fields.firstNameLastName.required}
            name="lastName"
            autoComplete="family-name"
          />
        </React.Fragment>
      ) : null}
      {fields.companyName?.enabled ? (
        <TextField
          value={values.companyName}
          onChange={onChange}
          label={t('personal_details.company_name')}
          placeholder={t('personal_details.company_name')}
          error={!!errors.companyName}
          helperText={errors.companyName}
          required={fields.companyName.required}
          name="companyName"
          autoComplete="organization"
        />
      ) : null}
      {fields.address?.enabled ? (
        <React.Fragment>
          <TextField
            value={values.address}
            onChange={onChange}
            label={t('personal_details.address')}
            placeholder={t('personal_details.address')}
            error={!!errors.address}
            helperText={errors.address}
            required={fields.address.required}
            name="address"
            autoComplete="address-line1"
          />
          <TextField
            value={values.address2}
            onChange={onChange}
            label={t('personal_details.address2')}
            placeholder={t('personal_details.address2')}
            error={!!errors.address2}
            helperText={errors.address2}
            name="address2"
            autoComplete="address-line2"
          />
          <TextField
            value={values.city}
            onChange={onChange}
            label={t('personal_details.city')}
            placeholder={t('personal_details.city')}
            error={!!errors.city}
            helperText={errors.city}
            required={fields.address.required}
            name="city"
            autoComplete="address-level2"
          />
          <TextField
            value={values.state}
            onChange={onChange}
            label={t('personal_details.state')}
            placeholder={t('personal_details.state')}
            error={!!errors.state}
            helperText={errors.state}
            required={fields.address.required}
            name="state"
            autoComplete="address-level1"
          />
          <TextField
            value={values.postCode}
            onChange={onChange}
            label={t('personal_details.post_code')}
            placeholder={t('personal_details.post_code')}
            error={!!errors.postCode}
            helperText={errors.postCode}
            required={fields.address.required}
            name="postCode"
            autoComplete="postal-code"
          />
        </React.Fragment>
      ) : null}
      {fields.phoneNumber?.enabled ? (
        <TextField
          value={values.phoneNumber}
          onChange={onChange}
          label={t('personal_details.phone_number')}
          placeholder={t('personal_details.phone_number')}
          error={!!errors.phoneNumber}
          helperText={errors.phoneNumber}
          required={fields.phoneNumber.required}
          name="phoneNumber"
        />
      ) : null}
      {fields.birthDate?.enabled ? (
        <DateField
          value={values.birthDate}
          onChange={(event) => setValue('birthDate', event.currentTarget.value)}
          label={t('personal_details.birth_date')}
          error={!!errors.birthDate}
          helperText={errors.birthDate}
          required={fields.birthDate.required}
          name="birthDate"
        />
      ) : null}
      {questions.map((question) => renderQuestion(question))}
      <Button
        className={styles.continue}
        type="submit"
        label={t('personal_details.continue')}
        variant="contained"
        color="primary"
        size="large"
        disabled={submitting}
        fullWidth
      />
      {submitting && <LoadingOverlay transparentBackground inline />}
    </form>
  );
};

export default PersonalDetailsForm;
