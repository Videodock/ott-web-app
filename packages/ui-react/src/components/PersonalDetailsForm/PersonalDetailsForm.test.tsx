import { render } from '@testing-library/react';
import type { CustomFormField } from '@jwp/ott-common/types/account';

import PersonalDetailsForm from './PersonalDetailsForm';

const noop = () => {
  /**/
};
const values = {
  firstName: 'Vince',
  lastName: 'Guy',
  birthDate: '27-Feb-1985',
  companyName: 'JW Player',
  phoneNumber: '+1 555 555 5555',
};
const fields: CustomFormField[] = [
  {
    type: 'input',
    name: 'firstName',
    label: 'First name',
    placeholder: '',
    required: true,
    defaultValue: '',
  },
  {
    type: 'input',
    name: 'lastName',
    label: 'Last name',
    placeholder: '',
    required: true,
    defaultValue: '',
  },
  {
    type: 'input',
    name: 'companyName',
    label: 'Company name',
    placeholder: '',
    required: false,
    defaultValue: '',
  },
  {
    type: 'datepicker',
    name: 'birthDate',
    label: 'Birth date',
    placeholder: '',
    required: false,
    defaultValue: '',
  },
  {
    type: 'input',
    name: 'phoneNumber',
    label: 'Phone Number',
    placeholder: '',
    required: false,
    defaultValue: '',
  },
  {
    type: 'input',
    name: 'custom_1',
    label: 'This is a text field',
    placeholder: '',
    required: false,
    defaultValue: '',
  },
  {
    type: 'checkbox',
    name: 'custom_2',
    label: 'This is a checkbox field',
    options: { accept: 'accept' },
    placeholder: '',
    required: false,
    enabledByDefault: false,
  },
  {
    type: 'radio',
    name: 'custom_3',
    label: 'This is a radio field',
    options: { option1: 'Option 1', option2: 'Option 2' },
    placeholder: '',
    required: true,
    defaultValue: '',
  },
  {
    type: 'select',
    name: 'custom_4',
    label: 'This is a select field',
    options: { option1: 'Option 1', option2: 'Option 2', option3: 'Option 3' },
    placeholder: '',
    required: true,
    defaultValue: '',
  },
];

describe('<PersonalDetailsForm>', () => {
  test('Renders without crashing', () => {
    const { container } = render(<PersonalDetailsForm onSubmit={noop} onChange={noop} errors={{}} values={values} submitting={false} fields={fields} />);

    expect(container).toMatchSnapshot();
  });

  test('Renders with errors', () => {
    const { container } = render(
      <PersonalDetailsForm
        onSubmit={noop}
        onChange={noop}
        errors={{
          firstName: 'this is wrong',
          lastName: 'also this',
          birthDate: 'February 30th? Really?',
          phoneNumber: 'Invalid',
          custom_1: 'Please enter some text',
          custom_2: 'Check this box please!',
          custom_3: "It's only two choices",
          custom_4: 'Select something...',
          form: 'This is a form error',
        }}
        values={values}
        submitting={false}
        fields={fields}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
