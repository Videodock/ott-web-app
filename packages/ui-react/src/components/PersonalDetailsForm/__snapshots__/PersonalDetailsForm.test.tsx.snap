// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PersonalDetailsForm> > Renders with errors 1`] = `
<div>
  <form
    data-testid="personal_details-form"
    novalidate=""
  >
    <h2
      class="_title_8a9484"
    >
      personal_details.title
    </h2>
    <div
      aria-live="assertive"
      class="_formFeedback_99f247 _error_99f247"
    >
      This is a form error
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_firstname"
      >
        personal_details.fist_name
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_firstname"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_firstname"
          name="firstName"
          placeholder="personal_details.fist_name"
          required=""
          type="text"
          value="Vince"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_firstname_1235"
      >
        this is wrong
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_lastname"
      >
        personal_details.last_name
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_lastname"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_lastname"
          name="lastName"
          placeholder="personal_details.last_name"
          required=""
          type="text"
          value="Guy"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_lastname_1235"
      >
        also this
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_companyname"
      >
        personal_details.company_name
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_companyname"
          aria-invalid="false"
          aria-required="false"
          class="_input_859b20"
          id="text-field_1235_companyname"
          name="companyName"
          placeholder="personal_details.company_name"
          type="text"
          value="JW Player"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_companyname_1235"
      >
        So much errors
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_address"
      >
        personal_details.address
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_address"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_address"
          name="address"
          placeholder="personal_details.address"
          required=""
          type="text"
          value="123 Nowhere St."
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_address_1235"
      >
        Do you know where you live
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_address2"
      >
        personal_details.address2
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_address2"
          aria-invalid="false"
          aria-required="false"
          class="_input_859b20"
          id="text-field_1235_address2"
          name="address2"
          placeholder="personal_details.address2"
          type="text"
          value="Apt 1W"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_address2_1235"
      >
        Come on
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_city"
      >
        personal_details.city
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_city"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_city"
          name="city"
          placeholder="personal_details.city"
          required=""
          type="text"
          value="Chicago"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_city_1235"
      >
        stop it
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_state"
      >
        personal_details.state
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_state"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_state"
          name="state"
          placeholder="personal_details.state"
          required=""
          type="text"
          value="IL"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_state_1235"
      >
        Confusion
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_postcode"
      >
        personal_details.post_code
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_postcode"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_postcode"
          name="postCode"
          placeholder="personal_details.post_code"
          required=""
          type="text"
          value="60601"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_postcode_1235"
      >
        Post Code should be 5 centimeters
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_phonenumber"
      >
        personal_details.phone_number
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20 _error_859b20"
      >
        <input
          aria-describedby="helper_text_1235_phonenumber"
          aria-invalid="true"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_phonenumber"
          name="phoneNumber"
          placeholder="personal_details.phone_number"
          required=""
          type="text"
          value="+1 555 555 5555"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_phonenumber_1235"
      >
        Invalid
      </div>
    </div>
    <div
      class="_formField_86f2f5 _dateField_0addad _error_0addad"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_birthdate"
      >
        personal_details.birth_date
      </label>
      <div
        class="_container_0addad"
      >
        <input
          aria-describedby="helper_text_1235_birthdate"
          aria-invalid="true"
          class="_hiddenInput_0addad"
          id="text-field_1235_birthdate-hidden"
          name="birthDate"
        />
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-date"
          maxlength="2"
          name="date"
          placeholder="dd"
          type="number"
          value="27"
        />
         / 
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-month"
          maxlength="2"
          name="month"
          placeholder="mm"
          type="number"
          value="2"
        />
         / 
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-year"
          maxlength="4"
          name="year"
          placeholder="yyyy"
          type="number"
          value="1985"
        />
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_birthdate_1235"
      >
        February 30th? Really?
      </div>
    </div>
    <div
      class="_formField_86f2f5"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_key1"
      >
        Which number is 2?
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_d97916 _error_d97916"
      >
        <select
          aria-describedby="helper_text_1235_key1"
          aria-invalid="true"
          aria-required="true"
          class="_select_d97916"
          id="text-field_1235_key1"
          name="key1"
          required=""
        >
          <option
            class="_option_d97916"
            disabled=""
            value=""
          >
            personal_details.select_answer
          </option>
          <option
            class="_option_d97916"
            value="1"
          >
            1
          </option>
          <option
            class="_option_d97916"
            value="2"
          >
            2
          </option>
          <option
            class="_option_d97916"
            value="3"
          >
            3
          </option>
        </select>
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_key1_1235"
      >
        question 1 is invalid
      </div>
    </div>
    <div
      class="_formField_86f2f5"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_key2"
      >
        Is this a question?
      </label>
      <div
        class="_radio_d50e6e"
      >
        <input
          aria-describedby="helper_text_1235_key2"
          id="text-field_1235_key20"
          name="key2"
          type="radio"
          value="Yes"
        />
        <label
          for="text-field_1235_key20"
        >
          Yes
        </label>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          aria-describedby="helper_text_1235_key2"
          checked=""
          id="text-field_1235_key21"
          name="key2"
          type="radio"
          value="No"
        />
        <label
          for="text-field_1235_key21"
        >
          No
        </label>
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_key2_1235"
      >
        Invalid answer
      </div>
    </div>
    <div
      class="_formField_86f2f5 _checkbox_a98c3a _error_a98c3a"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_ccc"
      >
        Check this off.
      </label>
      <div
        class="_row_a98c3a"
      >
        <input
          aria-describedby="helper_text_1235_ccc"
          aria-required="true"
          checked=""
          id="text-field_1235_ccc"
          name="ccc"
          type="checkbox"
          value="CheckMe"
        />
        <label
          for="text-field_1235_ccc"
        >
          <span
            role="presentation"
          >
            *
          </span>
          CheckMe
        </label>
      </div>
      <div
        aria-live="assertive"
        class="_helperText_da58e0 _error_da58e0"
        id="helper_text_1235_ccc_1235"
      >
        No way Jose
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _continue_8a9484 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      type="submit"
    >
      <span>
        personal_details.continue
      </span>
    </button>
  </form>
</div>
`;

exports[`<PersonalDetailsForm> > Renders without crashing 1`] = `
<div>
  <form
    data-testid="personal_details-form"
    novalidate=""
  >
    <h2
      class="_title_8a9484"
    >
      personal_details.title
    </h2>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_firstname"
      >
        personal_details.fist_name
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_firstname"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_firstname"
          name="firstName"
          placeholder="personal_details.fist_name"
          required=""
          type="text"
          value="Vince"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_lastname"
      >
        personal_details.last_name
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_lastname"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_lastname"
          name="lastName"
          placeholder="personal_details.last_name"
          required=""
          type="text"
          value="Guy"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_companyname"
      >
        personal_details.company_name
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_companyname"
          aria-invalid="false"
          aria-required="false"
          class="_input_859b20"
          id="text-field_1235_companyname"
          name="companyName"
          placeholder="personal_details.company_name"
          type="text"
          value="JW Player"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_address"
      >
        personal_details.address
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_address"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_address"
          name="address"
          placeholder="personal_details.address"
          required=""
          type="text"
          value="123 Nowhere St."
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_address2"
      >
        personal_details.address2
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_address2"
          aria-invalid="false"
          aria-required="false"
          class="_input_859b20"
          id="text-field_1235_address2"
          name="address2"
          placeholder="personal_details.address2"
          type="text"
          value="Apt 1W"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_city"
      >
        personal_details.city
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_city"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_city"
          name="city"
          placeholder="personal_details.city"
          required=""
          type="text"
          value="Chicago"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_state"
      >
        personal_details.state
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_state"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_state"
          name="state"
          placeholder="personal_details.state"
          required=""
          type="text"
          value="IL"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_postcode"
      >
        personal_details.post_code
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_postcode"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_postcode"
          name="postCode"
          placeholder="personal_details.post_code"
          required=""
          type="text"
          value="60601"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _textField_c799b0"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_phonenumber"
      >
        personal_details.phone_number
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_859b20"
      >
        <input
          aria-describedby="helper_text_1235_phonenumber"
          aria-invalid="false"
          aria-required="true"
          class="_input_859b20"
          id="text-field_1235_phonenumber"
          name="phoneNumber"
          placeholder="personal_details.phone_number"
          required=""
          type="text"
          value="+1 555 555 5555"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5 _dateField_0addad"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_birthdate"
      >
        personal_details.birth_date
      </label>
      <div
        class="_container_0addad"
      >
        <input
          aria-describedby="helper_text_1235_birthdate"
          aria-invalid="false"
          class="_hiddenInput_0addad"
          id="text-field_1235_birthdate-hidden"
          name="birthDate"
        />
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-date"
          maxlength="2"
          name="date"
          placeholder="dd"
          type="number"
          value="27"
        />
         / 
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-month"
          maxlength="2"
          name="month"
          placeholder="mm"
          type="number"
          value="2"
        />
         / 
        <input
          class="_input_0addad"
          id="text-field_1235_birthdate-year"
          maxlength="4"
          name="year"
          placeholder="yyyy"
          type="number"
          value="1985"
        />
      </div>
    </div>
    <div
      class="_formField_86f2f5"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_key1"
      >
        Which number is 2?
        <span
          role="presentation"
        >
          *
        </span>
      </label>
      <div
        class="_container_d97916"
      >
        <select
          aria-describedby="helper_text_1235_key1"
          aria-invalid="false"
          aria-required="true"
          class="_select_d97916"
          id="text-field_1235_key1"
          name="key1"
          required=""
        >
          <option
            class="_option_d97916"
            disabled=""
            value=""
          >
            personal_details.select_answer
          </option>
          <option
            class="_option_d97916"
            value="1"
          >
            1
          </option>
          <option
            class="_option_d97916"
            value="2"
          >
            2
          </option>
          <option
            class="_option_d97916"
            value="3"
          >
            3
          </option>
        </select>
      </div>
    </div>
    <div
      class="_formField_86f2f5"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_key2"
      >
        Is this a question?
      </label>
      <div
        class="_radio_d50e6e"
      >
        <input
          aria-describedby="helper_text_1235_key2"
          id="text-field_1235_key20"
          name="key2"
          type="radio"
          value="Yes"
        />
        <label
          for="text-field_1235_key20"
        >
          Yes
        </label>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          aria-describedby="helper_text_1235_key2"
          checked=""
          id="text-field_1235_key21"
          name="key2"
          type="radio"
          value="No"
        />
        <label
          for="text-field_1235_key21"
        >
          No
        </label>
      </div>
    </div>
    <div
      class="_formField_86f2f5 _checkbox_a98c3a"
    >
      <label
        class="_label_86f2f5"
        for="text-field_1235_ccc"
      >
        Check this off.
      </label>
      <div
        class="_row_a98c3a"
      >
        <input
          aria-describedby="helper_text_1235_ccc"
          aria-required="true"
          checked=""
          id="text-field_1235_ccc"
          name="ccc"
          type="checkbox"
          value="CheckMe"
        />
        <label
          for="text-field_1235_ccc"
        >
          <span
            role="presentation"
          >
            *
          </span>
          CheckMe
        </label>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _continue_8a9484 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      type="submit"
    >
      <span>
        personal_details.continue
      </span>
    </button>
  </form>
</div>
`;
